src/apps/bridge/sensor_drivers/rbrCodaSensor.cpp

```diff
@@ -0,0 +1,194 @@
+#include "rbrCodaSensor.h"
+#include "app_config.h"
+#include "avgSampler.h"
+#include "bm_network.h"
+#include "bm_pubsub.h"
+#include "bm_rbr_data_msg.h"
+#include "bridgeLog.h"
+#include "device_info.h"
+#include "reportBuilder.h"
+#include "semphr.h"
+#include "stm32_rtc.h"
+#include "topology_sampler.h"
+#include "util.h"
+#include <new>
+
+bool RbrCodaSensor::subscribe() {
+  bool rval = false;
+  char *sub = static_cast<char *>(pvPortMalloc(BM_TOPIC_MAX_LEN));
+  configASSERT(sub);
+  int topic_strlen = snprintf(sub, BM_TOPIC_MAX_LEN, "%" PRIx64 "%s", node_id, subtag);
+  if (topic_strlen > 0) {
+    rval = bm_sub_wl(sub, topic_strlen, rbrCodaSubCallback);
+  }
+  vPortFree(sub);
+  return rval;
+}
+
+void RbrCodaSensor::rbrCodaSubCallback(uint64_t node_id, const char *topic, uint16_t topic_len,
+                                       const uint8_t *data, uint16_t data_len, uint8_t type,
+                                       uint8_t version) {
+  (void)type;
+  (void)version;
+  printf("RBR CODA data received from node %" PRIx64 " On topic: %.*s\n", node_id, topic_len,
+         topic);
+  RbrCoda_t *rbr_coda = static_cast<RbrCoda_t *>(sensorControllerFindSensorById(node_id));
+  if (rbr_coda && rbr_coda->type == SENSOR_TYPE_RBR_CODA) {
+    if (xSemaphoreTake(rbr_coda->_mutex, portMAX_DELAY)) {
+      static BmRbrDataMsg::Data rbr_data;
+      if (BmRbrDataMsg::decode(rbr_data, data, data_len) == CborNoError) {
+        char *log_buf = static_cast<char *>(pvPortMalloc(SENSOR_LOG_BUF_SIZE));
+        configASSERT(log_buf);
+        rbr_coda->temp_deg_c.addSample(rbr_data.temperature_deg_c);
+        rbr_coda->pressure_ubar.addSample(rbr_data.pressure_deci_bar);
+        rbr_coda->reading_count++;
+        // Update the RBR CODA's latest sensor type
+        // TODO - verify that we should do this here v.s. when we aggregate. Although I'm not sure if we will be able to do that in the aggregate function.
+        rbr_coda->latest_sensor_type = rbr_data.sensor_type;
```

Comment:
I'll change this in a separate PR based on the discussions Russ and I have had here: https://app.shortcut.com/sofar/story/198619/bridge-support-for-the-bm-rbr